src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -8779,19 +8779,18 @@ def create_run_step(self, step_run: StepRunRequest) -> StepRunResponse:
                 session.commit()
                 session.refresh(step_schema)
 
-            # Save parent step IDs into the database.
-            for parent_step_id in step_run.parent_step_ids:
-                self._set_run_step_parent_step(
-                    child_step_run=step_schema,
-                    parent_id=parent_step_id,
-                    session=session,
-                )
-
             session.commit()
             session.refresh(step_schema)
 
             step_model = step_schema.to_model(include_metadata=True)
 
+            for upstream_step in step_model.spec.upstream_steps:
+                self._set_run_step_parent_step(
+                    child_step_run=step_schema,
+                    parent_step_name=upstream_step,
+                    session=session,
+                )
+
```

Comment:
This is the only change I didn't understand. Did you move this after `to_model` to keep the step model clean of upstream steps?